src/mlpack/methods/ann/layer/layer_norm_impl.hpp

```diff
@@ -31,14 +31,67 @@ LayerNorm<InputDataType, OutputDataType>::LayerNorm() :
 
 template <typename InputDataType, typename OutputDataType>
 LayerNorm<InputDataType, OutputDataType>::LayerNorm(
-    const size_t size, const double eps) :
+    const size_t size, const double eps):
     size(size),
     eps(eps),
     loading(false)
 {
   weights.set_size(size + size, 1);
 }
 
+
+template<typename InputDataType, typename OutputDataType>
+LayerNorm<InputDataType, OutputDataType>::LayerNorm(
+    const LayerNorm& layer):
+    size(layer.size),
+    eps(layer.eps),
+    weights(layer.weights),
+    loading(true)
+{
+    // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+LayerNorm<InputDataType, OutputDataType>::LayerNorm(
+    LayerNorm&& layer):
+    size(0),
+    eps(1e-8),
+    weights(std::move(layer.weights)),
+    loading(false)
+{
+    // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+LayerNorm<InputDataType, OutputDataType>&
+LayerNorm<InputDataType, OutputDataType>::
+operator=(const LayerNorm& layer)
+{
+    if (this != &layer)
+    {
+        size = layer.size;
+        eps = layer.eps;
+        weights = layer.weights;
+        loading = true;
+    }
+    return *this;
+}
+
+template<typename InputDataType, typename OutputDataType>
+LayerNorm<InputDataType, OutputDataType>&
+LayerNorm<InputDataType, OutputDataType>::
+operator=(LayerNorm&& layer)
+{
+    if (this != &layer)
+    {
+        size = 0;
+        eps = 1e-8;
```

Comment:
Should be `move(layer.size) and move(layer.eps)`.

Reply:
Yes about that...I am a bit confused about the following two implementations [linear_impl.hpp (Line (92-93)](https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/linear_impl.hpp#L92) and [linear3d_impl.hpp (Line(91-92)](https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/linear3d_impl.hpp#L91). Is there any particular reason for this difference? 
One more question...why shouldn't `move(layer.size) and move(layer.eps).` be used in the move constructors?